contrib/win32/win32compat/shell-host.c

```diff
@@ -1595,6 +1597,21 @@ wmain(int ac, wchar_t **av)
 		exit(255);
 	}
 
+	/* assign to job object */
+	if ((job = CreateJobObjectW(NULL, NULL)) == NULL) {
+		printf_s("cannot create job object, error: %d", GetLastError());
+		return -1;
+	}
+
+	memset(&job_info, 0, sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION));
+	job_info.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+
+	if (!SetInformationJobObject(job, JobObjectExtendedLimitInformation, &job_info, sizeof(job_info)) ||
+		!AssignProcessToJobObject(job, GetCurrentProcess())) {
+		printf_s("cannot associate job object: %d", GetLastError());
+		return -1;
```

Comment:
I assume the handle is closed when process exits.

Reply:
Right.